lib/rest/include/hse/rest/params.h

```diff
@@ -0,0 +1,43 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (C) 2022 Micron Technology, Inc.  All rights reserved.
+ */
+
+#ifndef HSE_REST_PARAMS_H
+#define HSE_REST_PARAMS_H
+
+#include <stdbool.h>
+#include <stddef.h>
+
+#include <hse/error/merr.h>
+
+#define rest_params_get(params, param, value, def) _Generic((value), \
+        bool *: rest_params_get_bool, \
+        size_t *: rest_params_get_size, \
+        const char **: rest_params_get_string \
+    )(params, param, value, def)
```

Comment:
I intentionally left it lower case to make it seem like a regular function call.

Reply:
Use the time-honored all-caps convention to alert the reader that something unsavory may be happening here and to use caution...

Reply:
Look in `tgmath.h`. Nothing is uppercased.

Reply:
I realize lower case is easier to read, but the point of the convention is to make it obvious that it's not a function call...  Just because other people defy conventions doesn't mean it's a good idea...

Reply:
inline function?

Reply:
Doesn't work with `_Generic`